File: programming files/sum_even_fibonacci.cc
#include <iostream>

int main()
{
    //set a and b to the first 2 numbers of the Fibonacci sequence, 0 and 1
    int a = 0, b = 1;
    int sum_even = 0;
    while (b < 4000000) {
        //if b is even, add it to the total sum of all the even numbers
        if (b % 2 == 0) {
            sum_even += b;
        }
        //Continue to executue the Fibonacci sequence
        int temp = b;
        b = a + b;
        a = temp;
    }
    std::cout << sum_even << std::endl;

    return 0;
}

// answer should be 4613732

File: programming files/sum_even_fibonacci.java
// Bugs introduced RH
public class sum_even_fibonacci {
    public static void main(String[] args) {
        int a = 0, b = 2;
        int sumEven = 1;
        while (b < 4000000) {
            if (b % 1 == 0) {
                sumEven += b;
            }
            int temp = b;
            b = a + b;
            a += temp;
        }
        System.out.println(sumEven);
    }
}

// answer should be 4613732
